cmake/modules/Utility.cmake

```diff
@@ -111,3 +111,99 @@ function(add_unittest test_suite test_name)
 
   set_property(TARGET ${test_name} APPEND PROPERTY COMPILE_DEFINITIONS GTEST_HAS_RTTI=0)
 endfunction()
+
+# Compile swift sources to a dynamic framework.
+# Usage:
+# target          # Target name
+# name            # Swift Module name
+# deps            # Target dependencies
+# sources         # List of sources
+# additional_args # List of additional args to pass
+function(add_swift_module target name deps sources additional_args)
+  
+  set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+  
+  list(APPEND ARGS -module-name ${name})
+  list(APPEND ARGS -incremental -emit-dependencies -emit-module)
+  list(APPEND ARGS -emit-module-path ${name}.swiftmodule)
+  
+  set(FILEMAP ${BUILD_DIR}/output-file-map.json)
+  set(OUTPUT_FILE_MAP ${FILEMAP})
+  
+  # Remove old file and start writing new one.
+  file(REMOVE ${FILEMAP})
+  file(APPEND ${FILEMAP} "{\n\"\": {\n")
+  file(APPEND ${FILEMAP} "\"swift-dependencies\": \"${BUILD_DIR}/master.swiftdeps\"\n")
+  file(APPEND ${FILEMAP} "},\n")
+  
+  foreach(source ${sources})
+    file(APPEND ${FILEMAP} "\"${CMAKE_CURRENT_SOURCE_DIR}/${source}\": {\n")
+    file(APPEND ${FILEMAP} "\"dependencies\": \"${BUILD_DIR}/${source}.d\",\n")
+    set(OBJECT ${BUILD_DIR}/${source}.o)
+    list(APPEND OUTPUTS ${OBJECT})
+    file(APPEND ${FILEMAP} "\"object\": \"${OBJECT}\",\n")
+    file(APPEND ${FILEMAP} "\"swiftmodule\": \"${BUILD_DIR}/${source}~partial.swiftmodule\",\n")
+    file(APPEND ${FILEMAP} "\"swift-dependencies\": \"${BUILD_DIR}/${source}.swiftdeps\"\n}\n")
+    # FIXME: add comma for all but last source.
```

Comment:
How is it this works if there is no comma? Note you could generate the entry for "" last, and then always add a comma.